mergebot_poller.py

```diff
@@ -1,47 +1,126 @@
-"""TODO(jasonkuster): DO NOT SUBMIT without one-line documentation for mergebot_poller.
-
-TODO(jasonkuster): DO NOT SUBMIT without a detailed description of mergebot_poller.
+"""
+mergebot_poller is a utility which polls an SCM and follows commands.
 """
 
-import mergebot
+from github_helper import GithubHelper
+import merge
 import Queue
+from threading import Thread
 
-GITHUB_SECRET = '../../github_auth/apache-beam.secret'
+# TODO: Fetch authorized users from somewhere official.
+AUTHORIZED_USERS = ["davorbonaci"]
+BOT_NAME = 'merge-bot'
+ORG = 'apache'
 
 class MergebotPoller:
-  scm_type = ""
-  repository = ""
-  branch = ""
-  scm_pollers = { "github": poll_github }
-  scm_mergers = { "github": mergebot.merge_git }
-  work_queue = Queue.Queue()
+  """
+  MergebotPoller is a class which handles the polling of a configured SCM.
```

Comment:
I think MergebotPoller should be an interface or base class, and GithubPoller should be an implementation of that. Rather than having the poll method do this delegation, rely on the object hierarchy to take care of it.